packages/guides-graphs/resources/config/guides-graphs.php

```diff
@@ -3,15 +3,37 @@
 declare(strict_types=1);
 
 use phpDocumentor\Guides\Graphs\Directives\UmlDirective;
-use phpDocumentor\Guides\RestructuredText\Directives\BaseDirective;
+use phpDocumentor\Guides\Graphs\Nodes\UmlNode;
+use phpDocumentor\Guides\Graphs\Renderer\DiagramRenderer;
+use phpDocumentor\Guides\Graphs\Renderer\PlantumlRenderer;
+use phpDocumentor\Guides\Graphs\Renderer\PlantumlServerRenderer;
+use phpDocumentor\Guides\Graphs\Twig\UmlExtension;
+use phpDocumentor\Guides\NodeRenderers\TemplateNodeRenderer;
 use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
 
 return static function (ContainerConfigurator $container): void {
     $container->services()
         ->defaults()
         ->autowire()
         ->autoconfigure()
-        ->instanceof(BaseDirective::class)
+        ->set(UmlDirective::class)
         ->tag('phpdoc.guides.directive')
-        ->set(UmlDirective::class);
+
+        ->set('phpdoc.guides.', TemplateNodeRenderer::class)
+        ->tag('phpdoc.guides.noderenderer.html')
+        ->arg('$template', 'body/uml.html.twig')
+        ->arg('$nodeClass', UmlNode::class)
+
+        ->set(PlantumlRenderer::class)
+        ->arg('$plantUmlBinaryPath', '%guides.graphs.plantuml_binary%')
+
+        ->set(PlantumlServerRenderer::class)
+        ->arg(
+            '$plantumlServerUrl',
+            '%guides.graphs.plantuml_server%',
+        )
+        ->alias(DiagramRenderer::class, PlantumlServerRenderer::class)
```

Comment:
For now this is not configurable, the configuration options are there, but I need to add a service locator that allows you to register any class.